clang/include/clang/AST/Type.h

```diff
@@ -1982,9 +1986,9 @@ class alignas(8) Type : public ExtQualsTypeCommonBase {
   Type(TypeClass tc, QualType canon, TypeDependence Dependence)
       : ExtQualsTypeCommonBase(this,
                                canon.isNull() ? QualType(this_(), 0) : canon) {
-    static_assert(sizeof(*this) <= 8 + sizeof(ExtQualsTypeCommonBase),
+    static_assert(sizeof(*this) <= 16 + sizeof(ExtQualsTypeCommonBase),
                   "changing bitfields changed sizeof(Type)!");
```

Comment:
Can you explain why you think changing the size of `Type` would have no impact?

Reply:
This check has been broken, because `sizeof` didn't take into the account that `Type` objects are actually laid out in memory on 16-byte boundary, which results in 24 bytes of data and 8 bytes of padding. My change just affirms the status quo that Type is indeed takes 16 bytes more memory than `ExtQuals`, which has size 16.

Reply:
Would it make sense to change this to use `alignof(decltype(*this))` rather than `16` to make the connection to alignment more obvious?

Reply:
Good suggestion!